rascaline/src/systems/mod.rs

```diff
@@ -69,6 +69,7 @@ pub trait SystemBase: Send + Sync {
     /// distance between atoms is actually bellow the cutoff passed in the last
     /// call to `compute_neighbors`. This function is only valid to call after a
     /// call to `compute_neighbors`.
+    /// TODO or CHECK: is there a risk of double counting acoss pbc?
```

Comment:
Pairs are supposed to only be counted once, as `i - j` and not as `j - i`.
Is there a risk that going across boundaries (and potentially with other edge-cases at the same time) could cause a pair to be counted twice as `i - j[+1]` and `j - i[-1]` (especially if `i == j`) ?

Reply:
Right, I'm looking into something similar right now, I'll triple check that. But pairs between an atom and it's image should only be produced once per direction (i.e. you'll get a pair between the atoms in the central image and the (1 1 0) image, but no pair between the atom in the central image and the (-1 -1 0) image.

Reply:
I just wrote this test in systems/neighbors.rs, and it fails:
```rust
    #[test]
    fn small_cell_large_cutoffs() {
        let cell = UnitCell::cubic(0.5);
        let positions = [Vector3D::new(0.0, 0.0, 0.0)];
        let neighbors = NeighborsList::new(&positions, cell, 0.6);

        let expected = [
            (Vector3D::new(0.0, 0.0, 0.5), [0, 0, 1]),
            (Vector3D::new(0.0, 0.5, 0.0),  [0, 1, 0]),
            (Vector3D::new(0.5, 0.0, 0.0),  [1, 0, 0]),
        ];
        
        assert_eq!(neighbors.pairs.len(), 3);
        for (pair, (vector, shifts)) in neighbors.pairs.iter().zip(&expected) {
            assert_eq!(pair.first, 0);
            assert_eq!(pair.second, 0);
            assert_ulps_eq!(pair.distance, 0.5);
            assert_ulps_eq!(pair.vector, vector);
            assert_eq!(&pair.cell_shift_indices, shifts);
        }
    }
 ```
and this one (which I think is the wrong result) succeeds:
```rust
    #[test]
    fn small_cell_large_cutoffs() {
        let cell = UnitCell::cubic(0.5);
        let positions = [Vector3D::new(0.0, 0.0, 0.0)];
        let neighbors = NeighborsList::new(&positions, cell, 0.6);

        let expected = [
            (Vector3D::new(-0.5, 0.0, 0.0),  [-1, 0, 0]),
            (Vector3D::new(0.0, -0.5, 0.0),  [0, -1, 0]),
            (Vector3D::new(0.0, 0.0, -0.5), [0, 0, -1]),
            (Vector3D::new(0.0, 0.0, 0.5), [0, 0, 1]),
            (Vector3D::new(0.0, 0.5, 0.0),  [0, 1, 0]),
            (Vector3D::new(0.5, 0.0, 0.0),  [1, 0, 0]),
        ];
        
        assert_eq!(neighbors.pairs.len(), 6);
        for (pair, (vector, shifts)) in neighbors.pairs.iter().zip(&expected) {
            assert_eq!(pair.first, 0);
            assert_eq!(pair.second, 0);
            assert_ulps_eq!(pair.distance, 0.5);
            assert_ulps_eq!(pair.vector, vector);
            assert_eq!(&pair.cell_shift_indices, shifts);
        }
    }
```

Reply:
if the behaviour for the first one is the correct one, then the latest commit (https://github.com/Luthaf/rascaline/pull/240/commits/4c6354a41531a3b0820ca9e489edf908d2402ab0) fixes this (it can be cherry-picked as well)

Reply:
ok, that does look like a bug! Thanks for looking into it =) I don't have much time right now to fully evaluate the consequences, but I'll cherry-pick your commit asap!